src/state/slices/txHistorySlice/selectors.ts

```diff
@@ -140,21 +141,25 @@ export const selectTxIdsByFilter = createCachedSelector(
   selectAccountIdParamFromFilter,
   selectAssetIdParamFromFilter,
   selectTxStatusParamFromFilter,
-  (txIds, txs, data, accountIdFilter, assetIdFilter, txStatusFilter): TxId[] => {
-    // filter by accountIdFilter, if it exists, otherwise data for all accountIds
-
-    const filtered = pickBy(data, (_, accountId) => {
-      if (accountIdFilter) return accountId === accountIdFilter
-      return true
-    })
-    const flattened = values(filtered)
+  selectIsRfoxTxParamFromFilter,
+  (txIds, txs, data, accountIdFilter, assetIdFilter, txStatusFilter, isRfoxTxFilter): TxId[] => {
+    const maybeFilteredByAccountId = accountIdFilter
+      ? pickBy(data, (_, accountId) => {
+          return accountId === accountIdFilter
+        })
+      : data
+    const flattened = values(maybeFilteredByAccountId)
       .flatMap(byAssetId => (assetIdFilter ? byAssetId?.[assetIdFilter] : values(byAssetId).flat()))
       .filter(isSome)
     const uniqueIds = uniq(flattened)
-    const uniqueIdsByStatus = txStatusFilter
-      ? uniqueIds.filter(txId => txs[txId].status === txStatusFilter)
+    const maybeFilteredByRfox = isRfoxTxFilter
+      ? uniqueIds.filter(txId => txs[txId].data?.parser === 'rfox')
```

Comment:
preferably-blocking: we might want to leverage a filter by parser instead of bringing rfox specific filters there

Reply:
akschually let's do it - hell yeah generalization 032f91d3e7

also using this opened comment to comment re: useless arrow 3ad8c25f865e8779efff11f3254cb67592fdce67